src/constructs/loadbalancing/elb.ts

```diff
@@ -38,7 +38,8 @@ export class GuApplicationTargetGroup extends ApplicationTargetGroup {
   constructor(scope: GuStack, id: string, props: GuApplicationTargetGroupProps) {
     super(scope, id, props);
 
-    if (props.overrideId) (this.node.defaultChild as CfnTargetGroup).overrideLogicalId(id);
+    if (props.overrideId || (scope.migrated && props.overrideId !== false))
+      (this.node.defaultChild as CfnTargetGroup).overrideLogicalId(id);
```

Comment:
I'm not 100% whether the `GuApplicationTargetGroup` and `GuApplicationListenerProps` need to implement this or not.

Reply:
Neither am I! 

However, generally speaking when a team migrates I expect that they will preview the change set in the console. Lots of `Delete` lines are going to look pretty intimidating and make them question whether the migration is going to put their application at risk of suffering downtime.

Consequently I think it's best to keep the same resources during a migration wherever possible.

Reply:
Good point. Let's leave this then.